[QuestionAnswering] Move stable version from feature branch to main

rokulka opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

rokulka committed:
move stable to main

tjprescott reviewed (changes requested):
Holding this PR pending @KrzysztofCwalina's review.

rokulka committed:
fix validation failures

rokulka committed:
update operationIds

rokulka committed:
update model names

heaths reviewed (approved):
LGTM, though we probably want to add `x-ms-client-name` for a bunch of property renames as well. Using snake_case from Python:

- [x] AnswerSpan.confidence_score -> AnswerSpan.confidence
- [x] AnswerSpanRequest.confidence_score_threshold -> ShortAnswerOptions.confidence_threshold
- [x] AnswerSpanRequest.top_answers_with_span -> ShortAnswerOptions.top
- [x] KnowledgeBaseAnswer.answer_span -> KnowledgeBaseAnswer.short_answer
- [x] KnowledgeBaseAnswer.id -> KnowledgeBaseAnswer.qna_id
- [x] KnowledgeBaseAnswer.confidence_score -> KnowledgeBaseAnswer.confidence
- [x] KnowledgeBaseAnswerRequestContext.previous_user_query -> KnowledgeBaseAnswerContext.previous_question
- [x] QueryKnowledgeBaseOptions.confidence_score_threshold -> AnswersOptions.confidence_threshold
- [x] QueryKnowledgeBaseOptions.answer_span_request -> AnswersOptions.short_answer_options
- [x] QueryKnowledgeBaseOptions.ranker_type -> AnswersOptions.ranker_kind
- [x] QueryTextOptions.records -> AnswersFromTextOptions.text_documents
- [x] TextAnswer.confidence_score -> TextAnswer.confidence
- [x] TextAnswer.answer_span -> TextAnswer.short_answer

heaths committed:
Rename properties with x-ms-client-name

heaths committed:
Properly declare ShortAnswerOptions.enable as constant true

After https://github.com/Azure/autorest/pull/4184, modelAsString: false
must be added

heaths committed:
Fix linting error
